freeclimb/utils/signature_information.py

```diff
@@ -0,0 +1,30 @@
+import time, hmac, hashlib
+
+class SignatureInformation:
+    requestTimestamp = 0
+    signatures = []
+
+    def __init__(self, requestHeader):
+        signatureHeaders = requestHeader.split(",")
+        for signature in signatureHeaders:
+            header = signature.split("=")[0]
+            value = signature.split("=")[1]
+            if header == "t":
+                self.requestTimestamp = value
+            elif header == "v1":
+                self.signatures.append(value) 
+    
+    def isRequestTimeValid(self, tolerance):
+        currentTime = self.getCurrentUnixTime()
+        return int(self.requestTimestamp) + int(tolerance) < currentTime
```

Comment:
is there a reason we have to cast at all?

Reply:
If I remove the casting, it will not consider it a boolean function according to VSCode and causes issues with testing:

With casting:
![Screenshot 2023-03-30 at 11 30 30 AM](https://user-images.githubusercontent.com/118939810/228887413-257a9e04-5abb-499e-b37b-cd7adc378819.png)

Without:
![Screenshot 2023-03-30 at 11 31 09 AM](https://user-images.githubusercontent.com/118939810/228887611-0fc391e9-5843-45f4-8355-d9e1df1e0125.png)

Reply:
Can you add type hints? https://docs.python.org/3/library/typing.html

Reply:
will try this and make edits in latest commit